includes/class-mastodon-api.php

```diff
@@ -1489,25 +1458,35 @@ public function api_submit_post( $request ) {
 		return $this->get_status_array( get_post( $post_id ) );
 	}
 
-	public function api_get_media( $request ) {
-		$post_id = $request->get_param( 'post_id' );
-		if ( ! is_numeric( $post_id ) || $post_id < 0 ) {
+	/**
+	 * Get a Media_Attachment entity by ID.
+	 *
+	 * @param WP_REST_Request $request The full data about the request.
+	 * @return Entity\Media_Attachment|\WP_Error
+	 */
+	public function api_get_media( WP_REST_Request $request ) {
+		$attachment_id = $request->get_param( 'post_id' );
+		if ( ! is_numeric( $attachment_id ) || $attachment_id < 0 ) {
 			return new \WP_Error( 'mastodon_api_get_media', 'Invalid post ID', array( 'status' => 400 ) );
 		}
-		$attachment = wp_get_attachment_metadata( $post_id );
-		$post = get_post( $post_id );
-		return array(
-			'id'          => $post_id,
-			'type'        => 'image',
-			'url'         => wp_get_attachment_url( $post_id ),
-			'preview_url' => wp_get_attachment_url( $post_id ),
-			'height'      => $attachment['height'],
-			'width'       => $attachment['width'],
-			'description' => $post->post_excerpt,
-		);
+
+		/**
+		 * Filters the media attachment returned by the API.
+		 *
+		 * @param array $media_attachment The media attachment.
+		 * @param int   $attachment_id    The attachment ID.
+		 * @return Entity\Media_Attachment The media attachment.
+		 */
+		return apply_filters( 'mastodon_api_post_media_attachment', null, $attachment_id );
```

Comment:
I think we should verify that the filter returned a `Entity\Media_Attachment` otherwise a Mastodon app might crash.

Reply:
The `add_filter` in the handler is `mastodon_api_media_attachment`, we need to use either or the other.
```suggestion
		return apply_filters( 'mastodon_api_media_attachment', null, $attachment_id );
```